### ISSUE TITLE
Dispatch on spatial graph?

### ISSUE THREAD

**Issue by tpoisot** [gottacatchenall/EcoDynamics.jl#5]
If you move the type of this field

https://github.com/gottacatchenall/EcoDynamics.jl/blob/327b87b4fbf47b1ced4263dcbb90009fa81c4c6a/src/problem.jl#L6

into the type declaration, you can dispatch on whether or not it has a spatial graph.

**Comment by gottacatchenall** [comment]
yeah that's a vestigal component, I've moved to parameterizing w/ `Spatialness`, which will also enable dispatch against `Model{S<:Spatialness}` to replace the ugly `∂u_spatial` calls https://github.com/gottacatchenall/EcoDynamics.jl/blob/327b87b4fbf47b1ced4263dcbb90009fa81c4c6a/src/models/lvcompetition.jl#LL31C1-L31C20
